worlds/witness/hints.py

```diff
@@ -338,22 +340,93 @@ def get_priority_hint_locations(world: "WitnessWorld") -> List[str]:
+        k: v for k, v in world.multiworld.worlds[hint_loc.player].location_name_groups.items()
+        if hint_loc.name in v
+    }
+
+    locations_in_that_world = {
+        location.name for location in world.multiworld.get_locations(hint_loc.player) if location.address is not None
+    }
+
+    valid_location_groups: Dict[str, int] = {}
+
+    for group, locations in possible_location_groups.items():
+        present_locations = sum(location in locations_in_that_world for location in locations)
+        if present_locations > 1:
+            valid_location_groups[group] = present_locations
+
+    if valid_location_groups["Everywhere"] > 100 or world.options.vague_hints == "experimental":
+        del valid_location_groups["Everywhere"]
+
+    if world.options.vague_hints == "experimental":
+        parent_region = hint_loc.parent_region
+
+        # Assume that an "experimental" player would never want an "Everywhere" hint.
+        if not valid_location_groups:
+            return hint_loc.parent_region.name
+
+        # Assume that an "experimental" player enjoys the idea of weird region names, and thus
+        # make them possible even if location groups are available
+        if parent_region.name not in possible_location_groups:
+            parent_region_location_amount = sum(
+                location.address is not None for location in parent_region.locations
+            )
+            if parent_region.name != "Menu" and 2 < parent_region_location_amount < (len(locations_in_that_world) / 2):
+                valid_location_groups[hint_loc.parent_region.name] = parent_region_location_amount
+
+    if not valid_location_groups:
+        return "Everywhere"
+
+    location_groups_with_weights = {
+        # Listen. Just don't worry about it. :)))
+        location_group: (x ** 0.6) * math.e ** (- (x / 7) ** 0.6) if x > 6 else x / 6
+        for location_group, x in valid_location_groups.items()
+    }
+
+    location_groups = list(location_groups_with_weights.keys())
+    weights = list(location_groups_with_weights.values())
+
+    return world.random.choices(location_groups, weights, k=1)[0]
+
+
 def word_direct_hint(world: "WitnessWorld", hint: WitnessLocationHint):
     location_name = hint.location.name
     if hint.location.player != world.player:
         location_name += " (" + world.multiworld.get_player_name(hint.location.player) + ")"
 
     item = hint.location.item
     item_name = item.name
+
     if item.player != world.player:
         item_name += " (" + world.multiworld.get_player_name(item.player) + ")"
 
-    if hint.hint_came_from_location:
-        hint_text = f"{location_name} contains {item_name}."
-    else:
-        hint_text = f"{item_name} can be found at {location_name}."
+    hint_text = ""
+    area: Optional[str] = None
+
+    if world.options.vague_hints:
+        if hint.location.player == world.player:
+            area = try_getting_location_group_other_world(world, hint.location)
```

Comment:
This makes `try_getting_location_group_other_world` seem like it could use a rename.

Reply:
Lol that's true